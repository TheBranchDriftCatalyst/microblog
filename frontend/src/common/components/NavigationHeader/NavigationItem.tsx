import {
  NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuTrigger,
} from "@/common/ui/navigation-menu";
import { cn } from "@/lib/utils";
import React, { useMemo } from "react";

export const NavigationListItem = React.forwardRef<
  React.ElementRef<"a">,
  React.ComponentPropsWithoutRef<"a"> & { comp?: React.ElementType }
>(({ className, title, children, comp, ...props }, ref) => {
  const Comp = comp || "a";
  return (
    <li>
      <NavigationMenuLink asChild>
        <Comp
          ref={ref}
          className={cn(
            "block select-none space-y-1 rounded-md p-3 leading-none no-underline outline-none transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground",
            className,
          )}
          {...props}
        >
          <div className="text-sm font-medium leading-none">{title}</div>
          <p className="line-clamp-2 text-sm leading-snug text-muted-foreground">
            {children}
          </p>
        </Comp>
      </NavigationMenuLink>
    </li>
  );
});
NavigationListItem.displayName = "NavigationListItem";

export interface NavigationItemBaseProps {
  title: string;
}

export interface NavigationItemChildrenProps extends NavigationItemBaseProps {
  children?: React.ReactNode;
}

export interface NavigationItemListChildrenProps
  extends NavigationItemBaseProps {
  links?: NavigationListLinks[];
}

export interface NavigationListLinks {
  title: string;
  href: string;
  description: string;
}

export const _sampleLinkObjects: NavigationListLinks[] = Array.from(
  { length: 10 },
  (_, i) => ({
    title: `Link ${i + 1}`,
    href: "/",
    description: "A description of the link",
  }),
);

type NavigationItemProps = NavigationItemChildrenProps &
  NavigationItemListChildrenProps;

export const NavItemLinks = ({ links }: { links: NavigationListLinks[]}) => {
  return (
    <ul className="grid gap-3 p-4 md:w-[400px] lg:w-[500px] lg:grid-cols-[.75fr_1fr]">
      {links?.map(({ title, href, description }) => (
        <NavigationListItem key={title} href={href} title={title}>
          {description}
        </NavigationListItem>
      ))}
    </ul>
  );
};


export const NavigationItem = ({
  children,
  title,
  links,
}: NavigationItemProps) => {
  const navContent = useMemo(() => {
    if (children) {
      return children;
    }

    return <NavItemLinks links={links} />;
  }, [children, links]);

  return (
    <NavigationMenuItem>
      <NavigationMenuTrigger>{title}</NavigationMenuTrigger>
      <NavigationMenuContent>{navContent}</NavigationMenuContent>
    </NavigationMenuItem>
  );
};

export default NavigationItem;
