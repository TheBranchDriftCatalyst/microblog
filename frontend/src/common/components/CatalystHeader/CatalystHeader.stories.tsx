import { Meta } from '@storybook/react';
import React from "react";
import NavigationItem, { _sampleLinkObjects } from "../NavigationHeader/NavigationItem";
import Header from './CatalystHeader'; // Adjust the import path as needed

// Define the metadata for the story
const meta: Meta<typeof Header> = {
  title: 'Components/CatalystHeader',
  component: Header,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

export const Default = {
  args: {
    navigationItems: [
      <NavigationItem title="Components" links={_sampleLinkObjects} />,
      <NavigationItem title="Otherstuff" links={_sampleLinkObjects} />,
    ],
    title: "Catalyst Header",
    breadcrumbs: [
      { href: "/section1", name: "Section 1" },
      { href: "/section1/subsection1", name: "Subsection 1", compact: 3},
      { href: "/section1/subsection1/item1", name: "Item 1" },
      { href: "/section1/subsection1/item1/detail1", name: "Detail 1" },
      { href: "/section1/subsection1/item1/detail1/more", name: "More" },
    ]
  },
  // decorators: [clearPaddingDecorator], // Apply the custom decorator to this story
};
