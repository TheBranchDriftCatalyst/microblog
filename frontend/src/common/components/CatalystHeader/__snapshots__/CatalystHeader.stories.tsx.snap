// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/CatalystHeader Default smoke-test 1`] = `
<header class="sticky top-0 z-40 w-full border-b bg-background">
  <div class="ml-4 mr-4 flex justify-between">
    <div class="flex items-center">
      <div class="flex items-center">
        <p class="text-base sm:text-lg md:text-xl">
          Catalyst Header
        </p>
      </div>
    </div>
    <div class="flex items-center justify-around">
      <nav aria-label="Main"
           data-orientation="horizontal"
           dir="ltr"
           class="relative z-10 flex max-w-max flex-1 items-center justify-center"
      >
        <div style="position: relative;">
          <ul data-orientation="horizontal"
              class="group flex flex-1 list-none items-center justify-center space-x-1"
              dir="ltr"
          >
            <li class="relative">
              <button id="radix-:r0:-trigger-radix-:r1:"
                      data-state="closed"
                      aria-expanded="false"
                      aria-controls="radix-:r0:-content-radix-:r1:"
                      class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50 group"
                      data-radix-collection-item
              >
                Components
                <svg xmlns="http://www.w3.org/2000/svg"
                     width="24"
                     height="24"
                     viewbox="0 0 24 24"
                     fill="none"
                     stroke="currentColor"
                     stroke-width="2"
                     stroke-linecap="round"
                     stroke-linejoin="round"
                     class="lucide lucide-chevron-down relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
                     aria-hidden="true"
                >
                  <path d="m6 9 6 6 6-6">
                  </path>
                </svg>
              </button>
            </li>
            <li class="relative">
              <button id="radix-:r0:-trigger-radix-:r2:"
                      data-state="closed"
                      aria-expanded="false"
                      aria-controls="radix-:r0:-content-radix-:r2:"
                      class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-accent/50 data-[state=open]:bg-accent/50 group"
                      data-radix-collection-item
              >
                Otherstuff
                <svg xmlns="http://www.w3.org/2000/svg"
                     width="24"
                     height="24"
                     viewbox="0 0 24 24"
                     fill="none"
                     stroke="currentColor"
                     stroke-width="2"
                     stroke-linecap="round"
                     stroke-linejoin="round"
                     class="lucide lucide-chevron-down relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
                     aria-hidden="true"
                >
                  <path d="m6 9 6 6 6-6">
                  </path>
                </svg>
              </button>
            </li>
          </ul>
        </div>
        <div class="left-auto right-0 absolute top-full flex justify-center">
        </div>
      </nav>
    </div>
  </div>
</header>
`;
